polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1042,6 +1062,13 @@ where
 		.get(&relay_parent)
 		.map(|s| s.child("advertise-collation"));
 
+	let peer_data = state.peer_data.get_mut(&peer_id).ok_or(AdvertisementError::UnknownPeer)?;
+
+	if peer_data.version == CollationVersion::V1 && !state.active_leaves.contains_key(&relay_parent)
```

Comment:
Is this enough? The collator could still try to build an entire prospective parachain, by using the same relay parent. This is actually the most likely case. I don't see why a collator would use an older relay parent on purpose.

Reply:
> Is this enough? The collator could still try to build an entire prospective parachain, by using the same relay parent. This is actually the most likely case. I don't see why a collator would use an older relay parent on purpose.

`insert_advertisement` fails for v1 messages if collator attempts to build more than 1 collation.

Reply:
What we need to check is that the para head of the candidate is already included on the relay chain.

Reply:
Here we request persisted validation data from chain
https://github.com/paritytech/polkadot-sdk/blob/7ca2a26ddd36c5b44346fa3efea1bfe1cbbf24a7/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1828-L1834
And then validate candidate against it
https://github.com/paritytech/polkadot-sdk/blob/7ca2a26ddd36c5b44346fa3efea1bfe1cbbf24a7/polkadot/node/network/collator-protocol/src/validator_side/collation.rs#L147-L149

Reply:
Uh! Missed that part, thanks @slumber ! All good then!